### PR TITLE
Add kube package

### PR DESCRIPTION
**Pull request by stgleb**

### CODE REVIEW

File: pkg/kube/endpoint.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	service *KubeService
+}
+
+func (e *KubeEndpoint) GetKube(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	kubeId := vars["id"]
+
+	kube, err := e.service.Get(r.Context(), kubeId)
+
+	if err != nil {
```
Comment (yegorch) [inline pkg/kube/endpoint.go]: What if we have internal server error?

File: pkg/kube/endpoint.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	err = e.service.Create(r.Context(), kube)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
```
Comment (yegorch) [inline pkg/kube/endpoint.go]: Shouldn' it be accepted?

File: pkg/kube/endpoint_test.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	"net/http/httptest"
+	"testing"
+)
+
+func TestKubeProfileEndpointCreateProfile(t *testing.T) {
+	endpoint := &KubeEndpoint{}
+	kube := &Kube{
+		Name:              "",
+		KubernetesVersion: "1.8.7",
+		MasterPublicIP:    "12.34.56.78",
```
Comment (yegorch) [inline pkg/kube/endpoint_test.go]: Why a single public IP? It could have a lot of them.

File: pkg/kube/service.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/supergiant/supergiant/pkg/storage"
+)
+
+type KubeService struct {
+	prefix      string
+	kubeStorage storage.Interface
```
Comment (yegorch) [inline pkg/kube/service.go]: Just storage is enough.

File: pkg/kube/service.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/supergiant/supergiant/pkg/storage"
+)
+
+type KubeService struct {
+	prefix      string
```
Comment (yegorch) [inline pkg/kube/service.go]: Maybe the prefix could be /supergiant/kube ? We should revisit it...

File: pkg/storage/mock.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
+package storage
```
Comment (yegorch) [inline pkg/storage/mock.go]: We already have that mock, see testutils/mock.go

File: pkg/kube/endpoint.go [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"gopkg.in/asaskevich/govalidator.v8"
+)
+
+type KubeEndpoint struct {
```
Comment (saheienko) [inline pkg/kube/endpoint.go]: Endpoint (or Handler) would be enough. 
Usage:
```
k kube.KubeService
vs
k kube.Service
```

File: pkg/kube/service.go (hunk 5 above)
Comment (saheienko) [reply pkg/kube/service.go]: Nope, "supergiant" wil be set as a rootPath in a storage.

File: pkg/kube/endpoint.go (hunk 2 above)
Comment (stgleb) [reply pkg/kube/endpoint.go]: Ok, I agree

File: pkg/kube/endpoint.go (hunk 1 above)
Comment (stgleb) [reply pkg/kube/endpoint.go]: @yegorch I need your error package for that))

**Comment by yegorch** [comment]
LGTM

**Pull request merged**